samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -409,6 +410,65 @@ static int sock_set_tls_options(
 			return errno;
 		}
 	}
+
+	/* DTLS connection ID */
+	if (dtls_connection_id) {
+		err = setsockopt(fd, SOL_TLS, TLS_DTLS_CID, &dtls_connection_id,
+				 sizeof(dtls_connection_id));
+		if (err) {
+			mosh_error("Unable to set DTLS connection ID, errno %d", errno);
+			return errno;
+		}
+	}
+
+	return 0;
+}
+
+static int sock_get_dtls_status(int fd, bool session_cache, int dtls_connection_id)
+{
+	int err;
+	int status;
+	int len = sizeof(status);
+	char status_str[64];
+
+	if (session_cache) {
+		err = getsockopt(fd, SOL_TLS, TLS_DTLS_HANDSHAKE_STATUS, &status, &len);
+		if (err) {
+			mosh_error("Unable to get DTLS handshake status, errno %d", errno);
+			return -errno;
```

Comment:
Other functions return errno's as positive values, probably this should do the same.